packages/create-strapi-starter/utils/build-starter.js

```diff
@@ -0,0 +1,188 @@
+
+/**
+ * @param  {string} starterUrl Github url to starter project
+ */
+async function getRepoInfo(starterUrl) {
+  const repoInfo = GitUrlParse(starterUrl);
+  const { name, full_name, ref } = repoInfo;
+
+  const response = await fetch(`https://api.github.com/repos/${full_name}`);
+
+  if (!response.ok) {
+    throw Error(`Could not fetch the default branch`);
+  }
+
+  const { default_branch } = await response.json();
+
+  return {
+    name,
+    full_name,
+    ref,
+    default_branch,
+  };
+}
+
+/**
+ * @param  {string} starterUrl Github url for strapi starter
+ * @param  {string} tmpDir Path to temporary directory
+ */
+async function downloadGithubRepo(starterUrl, tmpDir) {
+  const { name, full_name, ref, default_branch } = await getRepoInfo(starterUrl);
+
+  const branch = ref ? ref : default_branch;
+
+  // Download from GitHub
+  const codeload = `https://codeload.github.com/${full_name}/tar.gz/${branch}`;
+  const response = await fetch(codeload);
+  if (!response.ok) {
+    throw Error(`Could not download the ${chalk.green(`${name}`)} repository`);
+  }
+
+  await new Promise(resolve => {
+    response.body.pipe(tar.extract({ strip: 1, cwd: tmpDir })).on('close', resolve);
+  });
+}
+
+/**
+ * @param  {string} filePath Path to starter.json file
+ */
+function readStarterJson(filePath) {
+  const data = fse.readFileSync(filePath);
+  return JSON.parse(data);
+}
+
+/**
+ * @param  {string} rootPath Path to the project directory
+ * @param  {string} projectName Name of the project
+ */
+function initPackageJson(rootPath, projectName) {
+  const packageManager = hasYarn ? 'yarn' : 'npm run';
+
+  fse.writeJson(join(rootPath, 'package.json'), {
+    name: projectName,
+    private: true,
+    version: '0.0.0',
+    scripts: {
+      'dev:backend': `cd backend && ${packageManager} develop`,
+      'dev:frontend': `wait-on http://localhost:1337/admin && cd frontend && ${packageManager} develop --open`,
+      develop: `concurrently "${packageManager} dev:backend" "${packageManager} dev:frontend"`,
+    },
+    dependencies: {
+      concurrently: '6.0.0',
+      'wait-on': '5.2.1',
+    },
+  });
+}
+
+/**
+ * @param  {string} path The directory path for install
+ */
+async function installWithLogs(path) {
+  const installPrefix = chalk.yellow('Installing dependencies:');
+  const loader = ora(installPrefix).start();
+  const logInstall = (chunk = '') => {
+    loader.text = `${installPrefix} ${chunk
+      .toString()
+      .split('\n')
+      .join(' ')}`;
+  };
+
+  const runner = runInstall(path);
+  runner.stdout.on('data', logInstall);
+  runner.stderr.on('data', logInstall);
+
+  await runner;
+
+  loader.stop();
+  console.log(`Dependencies installed ${chalk.green('successfully')}.`);
+}
+
+/**
+ * @param  {object} projectArgs projectName and starterUrl for the project
+ * @param  {object} program Commands for generating new application
+ */
+module.exports = async function buildStarter(projectArgs, program) {
```

Comment:
We can do like with the template option & handle some shortcuts instead of github url only I think. (in another PR)

supporting: 
- gatsby-catalog => strapi/strapi-starter-gatsby-catalog
- cmp/gatsby-catalog => comp/strapi-strater/gatsby-catalog
- & so on like same algo as the templates